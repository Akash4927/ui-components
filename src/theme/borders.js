import { kebabCase, forEach } from 'lodash';

export const borderRadius = {
  none: '0',
  soft: '2px',
  circle: '50%',
};

// Collects all theme border radii as SCSS vars.
export function themeBorderRadiiAsScss() {
  const themeBorderRadii = [];

  forEach(borderRadius, (value, name) => {
    themeBorderRadii.push(`$border-radius-${kebabCase(name)}: ${value}`);
  });

  return themeBorderRadii;
}
