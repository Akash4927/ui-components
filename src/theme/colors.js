import { isString, kebabCase, forEach } from 'lodash';

export const colors = {
  // uncategorized colors
  aliceBlue: '#f0f8ff',
  amethystSmoke: '#9292b7',
  athens: '#e2e2ec',
  athensGray: '#eeeef4',
  black: '#1a1a1a',
  blueHaze: '#c0c0d5',
  comet: '#5b5b88',
  cometDark: '#5a5a7b',
  doveGray: '#737373',
  eastBay: '#4a5d87',
  gallery: '#f4f4f4',
  gunpowder: '#3c3c5b',
  kimberly: '#7d7da8',
  kimberlyDark: '#656597',
  logan: '#b1b1cb',
  mischka: '#dfdfea',
  mulledWine: '#46466a',
  mystic: '#dae4ea',
  silver: '#ccc',
  turquoise: '#037aa9',
  white: '#fff',
  whiteLilac: '#f0f0fa',
  whiteSmoke: '#fafafc',
  promQL: {
    /**
     * GHColors theme by Avi Aryan (http://aviaryan.in)
     * Inspired by Github syntax coloring
     */
    comment: '#bbbbbb',
    string: '#e3116c',
    punctuation: '#393a34',
    entity: '#36acaa',
    metricName: '#2aa198',
    attrName: '#00a4db',
    function: '#dc322f',
    deleted: '#9a050f',
    tag: '#00009f',
    // Dropdown colors,
    salmon: '#ff7c7c',
  },
  // use these for new stuff!
  primary: {
    charcoal: '#32324b',
    lavender: '#8383ac',
  },
  accent: {
    blue: '#00d2ff',
    orange: '#ff4b19',
  },
  status: {
    // Use these when colorizing elements that should indicate a state or outcome, ie Alerts.
    success: '#38bd93', // green
    error: '#c7254e', // red
    warning: '#d4ab27', // amber
    info: '#049cd7', // blue
    disabled: '#9a9a9a',
  },
  // Used by PrometheusGraph component
  graphThemes: {
    blue: [
      '#c7e9b4',
      '#7ecdbb',
      '#1eb5eb',
      '#1d91bf',
      '#235fa9',
      '#253393',
      '#084181',
    ],
    purple: [
      '#c1d4e7',
      '#9fbddb',
      '#8d95c6',
      '#8282ab',
      '#89429e',
      '#800f7a',
      '#0b0533',
    ],
    mixed: [
      '#c7e9b4',
      '#c1d4e7',
      '#7ecdbb',
      '#9fbddb',
      '#1eb5eb',
      '#8d95c6',
      '#1d91bf',
      '#8282ab',
      '#235fa9',
      '#89429e',
      '#253393',
      '#800f7a',
      '#084181',
      '#0b0533',
    ],
  },
  // Third-party specific colors - not to be used in the theme!
  thirdParty: {
    // Google single-click login
    cornflowerBlue: '#4285f4',
    azure: '#3769bb',
  },
};

// Flattens and collects all theme colors as SCSS vars
export function themeColorsAsScss() {
  const ignoreKeys = ['graphThemes'];
  const themeColors = [];

  forEach(colors, (value, name) => {
    const colorPrefix = `$color-${kebabCase(name)}`;
    if (ignoreKeys.includes(name)) return;

    if (isString(value)) {
      themeColors.push(`${colorPrefix}: ${value}`);
    } else {
      forEach(value, (innerValue, subname) => {
        if (ignoreKeys.includes(subname)) return;
        themeColors.push(`${colorPrefix}-${kebabCase(subname)}: ${innerValue}`);
      });
    }
  });

  return themeColors;
}
