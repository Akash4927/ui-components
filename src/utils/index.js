import intersection from 'lodash/intersection';
import first from 'lodash/first';
import keys from 'lodash/keys';

// Return the value of a set of props against a lookup object.
// Helps when using "ternary" props, ie: <Text small bold />
// Third argument is the default value to return in the event of an `undefined`;
export const lookupValue = (props, lookup, _default) => {
  const matches = intersection(keys(props), keys(lookup));

  if (matches.length > 1) {
    throw new Error(`You have specified conflicting props: ${JSON.stringify(matches)}`);
  }

  return lookup[first(matches)] || _default;
};
