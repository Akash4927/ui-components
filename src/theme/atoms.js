import { darken } from 'polished';

import { colors } from './colors';

// component-specific
export const atoms = {
  Alert: {
    default: {
      color: colors.black,
      background: colors.gallery,
      borderColor: darken(0.15, colors.gallery),
    },
    info: {
      color: colors.white,
      background: colors.status.info,
      borderColor: darken(0.15, colors.status.info),
    },
    success: {
      color: colors.white,
      background: colors.status.success,
      borderColor: darken(0.15, colors.status.success),
    },
    warning: {
      color: colors.white,
      background: colors.status.warning,
      borderColor: darken(0.15, colors.status.warning),
    },
    error: {
      color: colors.white,
      background: colors.status.error,
      borderColor: darken(0.15, colors.status.error),
    },
  },
  Button: {
    default: {
      color: colors.black,
      background: colors.white,
      hoverColor: colors.purple800,
      hoverBackground: darken(0.05, colors.white),
    },
    primary: {
      color: colors.white,
      background: colors.turquoise,
      hoverColor: colors.white,
      hoverBackground: darken(0.05, colors.turquoise),
    },
    danger: {
      color: colors.white,
      background: colors.status.error,
      hoverColor: colors.white,
      hoverBackground: darken(0.05, colors.status.error),
    },
    disabled: {
      color: colors.doveGray,
      background: colors.gallery,
      hoverColor: colors.doveGray,
      hoverBackground: colors.gallery,
    },
  },
};
