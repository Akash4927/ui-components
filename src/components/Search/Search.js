import React from 'react';
import styled from 'styled-components';
import { map, includes, without, isEmpty, last, noop } from 'lodash';
import PropTypes from 'prop-types';

import { copyPropTypes } from '../../utils/compose';
import Input from '../Input';
import Dropdown from '../Dropdown';

import Term from './_SearchTerm';

const TermsContainer = styled.ul`
  list-style: none;
  display: flex;
  margin: 0;
  padding: 0;
  flex-wrap: wrap;
`;

const Icon = styled.i`
  line-height: 34px;
  padding: 0 8px;
`;

const SearchInput = styled.div`
  display: flex;
  flex-wrap: wrap;
  flex: 3;
  width: 100%;
`;

const Styled = component => styled(component)`
  position: relative;
  display: flex;
  background-color: ${props => props.theme.colors.white};
  border: 1px solid ${props => props.theme.colors.purple100};
  border-radius: ${props => props.theme.borderRadius.soft};

  div,
  input {
    border: 0;
  }

  ${Input}, ${Dropdown} {
    padding: 0;
    margin: 0;
  }

  ${Dropdown} {
    flex: 1;
    border-left: 1px solid ${props => props.theme.colors.gray200};

    .dropdown-popover {
      width: auto;
    }
  }

  ${Input} {
    flex: 2;
    width: 100%;

    input {
      padding: 0 8px;
      width: 100%;
    }

    input:focus {
      outline: none;
    }

    div {
      margin: 0;
    }
  }
`;

class Search extends React.PureComponent {
  addSearchTerm = (ev, value) => {
    let nextPinnedTerms = this.props.pinnedTerms;
    // only push unique values
    if (!includes(nextPinnedTerms, value)) {
      nextPinnedTerms = [...nextPinnedTerms, value];
      this.props.onPin(nextPinnedTerms);
    }
    this.props.onChange('', nextPinnedTerms);
  };

  removeSearchTerm = value => {
    const nextPinnedTerms = without(this.props.pinnedTerms, value);
    this.props.onChange(this.props.query, nextPinnedTerms);
  };

  handleInputKeyPress = ev => {
    if (ev.key === 'Enter' && this.props.query.length > 0) {
      ev.preventDefault();
      this.addSearchTerm(ev, this.props.query);
    } else if (ev.key === 'Backspace' && this.props.query === '') {
      ev.preventDefault();
      const term = last(this.props.pinnedTerms);
      if (term) {
        // Allow the user to edit the text of the last term instead of removing the whole thing.
        this.removeSearchTerm(term);
      }
    }
  };

  handleInputChange = ev => {
    this.props.onChange(ev.target.value, this.props.pinnedTerms);
  };

  handleInputBlur = ev => {
    // If the input loses focus, pin the search term. Skip if input is empty.
    if (this.props.query) {
      this.addSearchTerm(ev, this.props.query);
    }
    this.props.onBlur(ev);
  };

  handleFilterChange = (ev, value) => {
    this.props.onFilterSelect(value);
  };

  render() {
    const { className, filters, placeholder, query, pinnedTerms } = this.props;

    return (
      <div className={className}>
        <Icon className="fa fa-search" />
        <SearchInput>
          <TermsContainer>
            {map(pinnedTerms, term => (
              <Term key={term} term={term} onRemove={this.removeSearchTerm} />
            ))}
          </TermsContainer>
          <Input
            hideValidationMessage
            onChange={this.handleInputChange}
            value={query}
            onKeyDown={this.handleInputKeyPress}
            onBlur={this.handleInputBlur}
            onFocus={this.props.onFocus}
            inputRef={ref => {
              this.input = ref;
            }}
            placeholder={pinnedTerms.length === 0 ? placeholder : null}
          />
        </SearchInput>

        {!isEmpty(filters) && (
          <Dropdown
            items={filters}
            placeholder="Filters"
            onChange={this.handleFilterChange}
          />
        )}
      </div>
    );
  }
}

Search.propTypes = {
  /**
   * The initial value to use to populate the search text field.
   * Changes to this prop will be ignored after initial render.
   */
  query: PropTypes.string.isRequired,
  /**
   * The initial pinned terms of the search field.
   * Changes to this prop will be ignored after initial render.
   */
  pinnedTerms: PropTypes.arrayOf(PropTypes.string).isRequired,
  /**
   * Handler that runs when the text input changes.
   * Returns the text as first argument, and the list of pinned terms as the second.
   */
  onChange: PropTypes.func.isRequired,
  /**
   * A list of selectable filters to be rendered in a `<Dropdown />`.
   * When an option is clicked, the `value` is added to the search terms.
   * This array will be passed directly to the `items` prop of the `<Dropdown />.
   * If omitted, no dropdown will be rendered.
   */
  filters: PropTypes.arrayOf(
    PropTypes.shape({
      value: PropTypes.string,
      label: PropTypes.string,
    })
  ),
  /**
   * Text that will be passed to the search input as the placeholder.
   */
  placeholder: PropTypes.string,
  /**
   * Handler that runs when a search is pinned or unpinned.
   * Returns an array of the currently pinned terms.
   */
  onPin: PropTypes.func,
  /**
   * Handler that runs when an item from the search filter dropdown is selected
   * Returns the selected filter value.
   */
  onFilterSelect: PropTypes.func,
  onFocus: PropTypes.func,
  onBlur: PropTypes.func,
};

Search.defaultProps = {
  placeholder: '',
  filters: [],
  onPin: noop,
  onFilterSelect: noop,
  onFocus: noop,
  onBlur: noop,
};

export default copyPropTypes(Search, Styled(Search));
