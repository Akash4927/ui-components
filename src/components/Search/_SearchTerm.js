import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { noop } from 'lodash';

import { copyPropTypes } from '../../utils/compose';

const Styled = component => styled(component)`
  margin: 4px 0 4px 2px;
  padding: 4px;
  background-color: ${props => props.theme.colors.blue400};
  margin-left: 4px;
  border-radius: ${props => props.theme.borderRadius.soft};
  display: flex;

  i {
    padding-left: 4px;
    cursor: pointer;
  }
`;

class SearchTerm extends React.PureComponent {
  handleRemove = () => {
    this.props.onRemove(this.props.term, this.props.label);
  };
  render() {
    const { className, term, label } = this.props;
    return (
      <li className={`${className} search-term`}>
        <div className="search-term-text">{label || term}</div>
        <i onClick={this.handleRemove} className="fa fa-times remove-term" />
      </li>
    );
  }
}

SearchTerm.propTypes = {
  /**
   * The internal identifier for a term.
   * If no `label` is supplied, `term` will be rendered.
   */
  term: PropTypes.string.isRequired,
  /**
   * Display value that will be rendered.
   */
  label: PropTypes.string,
  /**
   * Handler that will run when a term is removed.
   * The `term` prop will be passed to the `onRemove` handler.
   */
  onRemove: PropTypes.func,
};

SearchTerm.defaultProps = {
  onRemove: noop,
};

export default copyPropTypes(SearchTerm, Styled(SearchTerm));
