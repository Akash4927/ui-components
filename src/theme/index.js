import { colors, themeColorsAsScss } from './colors';
import { fontSizes, fontFamilies, themeFontsAsScss } from './fonts';
import { borderRadius, themeBorderRadiiAsScss } from './borders';
import { layers, themeLayersAsScss } from './layers';
import { spacing, themeSpacingsAsScss } from './spacings';
import { overlayIconSize, textColor, themeMiscVarsAsScss } from './misc';
import { boxShadow } from './box-shadows';
import { atoms } from './atoms';

export function themeVarsAsScss() {
  const themeVariables = []
    .concat(themeColorsAsScss())
    .concat(themeLayersAsScss())
    .concat(themeFontsAsScss())
    .concat(themeBorderRadiiAsScss())
    .concat(themeSpacingsAsScss())
    .concat(themeMiscVarsAsScss());
  return `${themeVariables.join('; ')};`;
}

const theme = {
  // Component-specific
  atoms,

  // Box shadows
  boxShadow,

  // Borders
  borderRadius,

  // Colors
  colors,

  // Fonts
  fontFamilies,
  fontSizes,

  // z-index layers
  layers,

  // Spacings
  spacing,

  // Misc
  overlayIconSize,
  textColor,
};

export default theme;
