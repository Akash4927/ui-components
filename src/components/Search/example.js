import React from 'react';
import styled from 'styled-components';
import { map, filter, includes } from 'lodash';

import { Example } from '../../utils/example';

import Search from './Search';

const List = ({ items, text, terms }) => {
  const list =
    text.length > 0 || terms.length > 0
      ? filter(items, i => (text === i || includes(terms, i): true))
      : items;

  return <ul>{map(list, item => <li key={item}>{item}</li>)}</ul>;
};

const Styled = component => styled(component)`
  ${Example} {
    margin-bottom: 96px;
  }
`;

class SearchExample extends React.Component {
  state = {
    queries: {
      normal: { text: '', terms: [] },
      filters: { text: '', terms: [] },
    },
  };
  render() {
    const {
      queries: { normal, filters },
    } = this.state;
    return (
      <div className={this.props.className}>
        <Example>
          <h3>Normal</h3>
          <Search
            onChange={(text, terms) =>
              this.setState({ queries: { ...this.state.queries, normal: { text, terms } } })
            }
          />
          <List items={['red', 'green', 'blue']} terms={normal.terms} text={normal.text} />
        </Example>
        <Example>
          <h3>With filters</h3>
          <Search
            onChange={(text, terms) =>
              this.setState({ queries: { ...this.state.queries, filters: { text, terms } } })
            }
            filters={[{ value: 'red', label: 'Red' }, { value: 'blue', label: 'Blue' }]}
          />
          <List items={['red', 'green', 'blue']} text={filters.text} terms={filters.terms} />
        </Example>
        <Example>
          <h3>With an initial query and filters</h3>
          <Search
            initialPinnedTerms={['foo:bar', 'is:automated']}
            filters={[
              { value: 'is:automated', label: 'Automated' },
              { value: 'is:locked', label: 'Locked' },
            ]}
          />
        </Example>
      </div>
    );
  }
}

export default Styled(SearchExample);
