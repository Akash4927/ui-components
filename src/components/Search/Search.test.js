import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { createSpy } from 'expect';
import { noop } from 'lodash';

import { withTheme } from '../../utils/theme';

import Search from './Search';

describe('<Search />', () => {
  describe('snapshots', () => {
    it('renders empty', () => {
      const tree = renderer
        .create(withTheme(<Search query="" pinnedTerms={[]} onChange={noop} />))
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('renders a search string', () => {
      const tree = renderer
        .create(
          withTheme(
            <Search query="ui-server" pinnedTerms={[]} onChange={noop} />
          )
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('renders a group of terms', () => {
      const tree = renderer
        .create(
          withTheme(
            <Search
              query=""
              pinnedTerms={['deployments', 'default']}
              onChange={noop}
            />
          )
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('renders filters', () => {
      const tree = renderer
        .create(
          withTheme(
            <Search
              query=""
              pinnedTerms={[]}
              onChange={noop}
              filters={[
                { value: 'automated', label: 'Automated' },
                { value: 'locked', label: 'Locked' },
              ]}
            />
          )
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  it('returns search terms', () => {
    const spy = createSpy();
    const search = mount(
      withTheme(<Search query="" pinnedTerms={['a']} onChange={spy} />)
    );
    const $input = search.find('input');
    $input.simulate('change', { target: { value: 'myterm' } });
    expect(spy.calls[0].arguments).toEqual(['myterm', ['a']]);
  });
  it('calls the onFilterSelect prop with selected filter value', () => {
    const spy = createSpy();
    const search = mount(
      withTheme(
        <Search
          query=""
          pinnedTerms={[]}
          onChange={noop}
          onFilterSelect={spy}
          filters={[
            { value: 'automated', label: 'Automated' },
            { value: 'locked', label: 'Locked' },
          ]}
        />
      )
    );

    // Open the dialog pop up
    // TODO: make the dropdown do `display: none` instead of removing it from the dom?
    search
      .find('.dropdown-toggle')
      .first()
      .simulate('click');
    // select a filter
    search
      .find('.dropdown-item')
      .first()
      .simulate('click');
    expect(spy.calls[0].arguments).toEqual(['automated']);
  });
  it('clears terms', () => {
    const spy = createSpy();
    const search = mount(
      withTheme(<Search query="" pinnedTerms={['a', 'b']} onChange={spy} />)
    );
    // Remove the 'a' search term
    search
      .find('.remove-term')
      .first()
      .simulate('click');
    expect(spy.calls[0].arguments).toEqual(['', ['b']]);
  });
  it('should raise an error if you forget to provide an onFilterSelect callback', () => {
    const search = mount(
      withTheme(
        <Search
          query="ui-server"
          pinnedTerms={[]}
          onChange={noop}
          filters={[{ value: '1', label: 'one' }]}
        />
      )
    );
    search
      .find('.dropdown-toggle')
      .first()
      .simulate('click');

    expect(() =>
      // select a filter
      search
        .find('.dropdown-item')
        .first()
        .simulate('click')
    ).toThrow();
  });
});
