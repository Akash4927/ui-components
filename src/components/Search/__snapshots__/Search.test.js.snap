// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> snapshots renders a group of terms 1`] = `
.c8 {
  visibility: hidden;
  display: inline-block;
  margin-left: 4px;
  font-size: 22px;
  color: #c7254e;
}

.c7 {
  margin-top: 10px;
  margin-bottom: 10px;
  height: 36px;
}

.c9 {
  display: none;
  font-size: 14px;
  padding-left: 8px;
  visibility: hidden;
  color: inherit;
}

.c6 {
  padding: 8px;
}

.c6 input {
  display: inline-block;
  padding: 0 12px;
  line-height: 36px;
  box-shadow: none;
  border: 1px solid hsl(0,0%,45%);
  border-radius: 4px;
  height: 36px;
  box-sizing: border-box;
}

.c4 {
  margin: 4px 0 4px 2px;
  padding: 4px;
  background-color: hsl(191,100%,50%);
  margin-left: 4px;
  border-radius: 4px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c4 i {
  padding-left: 4px;
  cursor: pointer;
}

.c3 {
  list-style: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0;
  padding: 0;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c1 {
  line-height: 34px;
  padding: 0 8px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex: 3;
  -ms-flex: 3;
  flex: 3;
  width: 100%;
}

.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: hsl(0,0%,100%);
  border: 1px solid hsl(240,20%,90%);
  border-radius: 4px;
}

.c0 div,
.c0 input {
  border: 0;
}

.c0 .c5,
.c0 .Dropdown-dxAxgO {
  padding: 0;
  margin: 0;
}

.c0 .Dropdown-dxAxgO {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  border-left: 1px solid hsl(0,0%,80%);
}

.c0 .c5 {
  -webkit-flex: 2;
  -ms-flex: 2;
  flex: 2;
  width: 100%;
}

.c0 .c5 input {
  padding: 0 8px;
  width: 100%;
}

.c0 .c5 input:focus {
  outline: none;
}

.c0 .c5 div {
  margin: 0;
}

<div
  className="c0"
>
  <i
    className="fa fa-search c1"
  />
  <div
    className="c2"
  >
    <ul
      className="c3"
    >
      <li
        className="c4 search-term"
      >
        <div
          className="search-term-text"
        >
          deployments
        </div>
        <i
          className="fa fa-times remove-term"
          onClick={[Function]}
        />
      </li>
      <li
        className="c4 search-term"
      >
        <div
          className="search-term-text"
        >
          default
        </div>
        <i
          className="fa fa-times remove-term"
          onClick={[Function]}
        />
      </li>
    </ul>
    <div
      className="c5 c6"
    >
      <label
        htmlFor={undefined}
      />
      <div
        className="c7"
      >
        <input
          onBlur={[Function]}
          onChange={[Function]}
          onKeyDown={[Function]}
          value=""
        />
        <i
          className="fa fa-times-circle c8"
        />
      </div>
      <span
        className="c9"
      />
    </div>
  </div>
</div>
`;

exports[`<Search /> snapshots renders a search string 1`] = `
.c7 {
  visibility: hidden;
  display: inline-block;
  margin-left: 4px;
  font-size: 22px;
  color: #c7254e;
}

.c6 {
  margin-top: 10px;
  margin-bottom: 10px;
  height: 36px;
}

.c8 {
  display: none;
  font-size: 14px;
  padding-left: 8px;
  visibility: hidden;
  color: inherit;
}

.c5 {
  padding: 8px;
}

.c5 input {
  display: inline-block;
  padding: 0 12px;
  line-height: 36px;
  box-shadow: none;
  border: 1px solid hsl(0,0%,45%);
  border-radius: 4px;
  height: 36px;
  box-sizing: border-box;
}

.c3 {
  list-style: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0;
  padding: 0;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c1 {
  line-height: 34px;
  padding: 0 8px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex: 3;
  -ms-flex: 3;
  flex: 3;
  width: 100%;
}

.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: hsl(0,0%,100%);
  border: 1px solid hsl(240,20%,90%);
  border-radius: 4px;
}

.c0 div,
.c0 input {
  border: 0;
}

.c0 .c4,
.c0 .Dropdown-dxAxgO {
  padding: 0;
  margin: 0;
}

.c0 .Dropdown-dxAxgO {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  border-left: 1px solid hsl(0,0%,80%);
}

.c0 .c4 {
  -webkit-flex: 2;
  -ms-flex: 2;
  flex: 2;
  width: 100%;
}

.c0 .c4 input {
  padding: 0 8px;
  width: 100%;
}

.c0 .c4 input:focus {
  outline: none;
}

.c0 .c4 div {
  margin: 0;
}

<div
  className="c0"
>
  <i
    className="fa fa-search c1"
  />
  <div
    className="c2"
  >
    <ul
      className="c3"
    />
    <div
      className="c4 c5"
    >
      <label
        htmlFor={undefined}
      />
      <div
        className="c6"
      >
        <input
          onBlur={[Function]}
          onChange={[Function]}
          onKeyDown={[Function]}
          value="ui-server"
        />
        <i
          className="fa fa-times-circle c7"
        />
      </div>
      <span
        className="c8"
      />
    </div>
  </div>
</div>
`;

exports[`<Search /> snapshots renders empty 1`] = `
.c7 {
  visibility: hidden;
  display: inline-block;
  margin-left: 4px;
  font-size: 22px;
  color: #c7254e;
}

.c6 {
  margin-top: 10px;
  margin-bottom: 10px;
  height: 36px;
}

.c8 {
  display: none;
  font-size: 14px;
  padding-left: 8px;
  visibility: hidden;
  color: inherit;
}

.c5 {
  padding: 8px;
}

.c5 input {
  display: inline-block;
  padding: 0 12px;
  line-height: 36px;
  box-shadow: none;
  border: 1px solid hsl(0,0%,45%);
  border-radius: 4px;
  height: 36px;
  box-sizing: border-box;
}

.c3 {
  list-style: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0;
  padding: 0;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c1 {
  line-height: 34px;
  padding: 0 8px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex: 3;
  -ms-flex: 3;
  flex: 3;
  width: 100%;
}

.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: hsl(0,0%,100%);
  border: 1px solid hsl(240,20%,90%);
  border-radius: 4px;
}

.c0 div,
.c0 input {
  border: 0;
}

.c0 .c4,
.c0 .Dropdown-dxAxgO {
  padding: 0;
  margin: 0;
}

.c0 .Dropdown-dxAxgO {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  border-left: 1px solid hsl(0,0%,80%);
}

.c0 .c4 {
  -webkit-flex: 2;
  -ms-flex: 2;
  flex: 2;
  width: 100%;
}

.c0 .c4 input {
  padding: 0 8px;
  width: 100%;
}

.c0 .c4 input:focus {
  outline: none;
}

.c0 .c4 div {
  margin: 0;
}

<div
  className="c0"
>
  <i
    className="fa fa-search c1"
  />
  <div
    className="c2"
  >
    <ul
      className="c3"
    />
    <div
      className="c4 c5"
    >
      <label
        htmlFor={undefined}
      />
      <div
        className="c6"
      >
        <input
          onBlur={[Function]}
          onChange={[Function]}
          onKeyDown={[Function]}
          value=""
        />
        <i
          className="fa fa-times-circle c7"
        />
      </div>
      <span
        className="c8"
      />
    </div>
  </div>
</div>
`;

exports[`<Search /> snapshots renders filters 1`] = `
.c7 {
  visibility: hidden;
  display: inline-block;
  margin-left: 4px;
  font-size: 22px;
  color: #c7254e;
}

.c6 {
  margin-top: 10px;
  margin-bottom: 10px;
  height: 36px;
}

.c8 {
  display: none;
  font-size: 14px;
  padding-left: 8px;
  visibility: hidden;
  color: inherit;
}

.c5 {
  padding: 8px;
}

.c5 input {
  display: inline-block;
  padding: 0 12px;
  line-height: 36px;
  box-shadow: none;
  border: 1px solid hsl(0,0%,45%);
  border-radius: 4px;
  height: 36px;
  box-sizing: border-box;
}

.c13 {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  padding: 0 12px;
  cursor: pointer;
}

.c11 {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  padding: 0 12px;
  cursor: pointer;
  border-radius: 4px;
  background-color: hsl(0,0%,100%);
  border: 1px solid hsl(0,0%,45%);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c11 .c12 {
  padding: 0;
}

.c11 div:last-child {
  margin-left: auto;
}

.c14 {
  padding-left: 1em;
  float: right;
  line-height: 36px !important;
}

.c10 {
  height: 36px;
  line-height: 34px;
  position: relative;
  width: 256px;
  box-sizing: border-box;
}

.c3 {
  list-style: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0;
  padding: 0;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c1 {
  line-height: 34px;
  padding: 0 8px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex: 3;
  -ms-flex: 3;
  flex: 3;
  width: 100%;
}

.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: hsl(0,0%,100%);
  border: 1px solid hsl(240,20%,90%);
  border-radius: 4px;
}

.c0 div,
.c0 input {
  border: 0;
}

.c0 .c4,
.c0 .c9 {
  padding: 0;
  margin: 0;
}

.c0 .c9 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  border-left: 1px solid hsl(0,0%,80%);
}

.c0 .c4 {
  -webkit-flex: 2;
  -ms-flex: 2;
  flex: 2;
  width: 100%;
}

.c0 .c4 input {
  padding: 0 8px;
  width: 100%;
}

.c0 .c4 input:focus {
  outline: none;
}

.c0 .c4 div {
  margin: 0;
}

<div
  className="c0"
>
  <i
    className="fa fa-search c1"
  />
  <div
    className="c2"
  >
    <ul
      className="c3"
    />
    <div
      className="c4 c5"
    >
      <label
        htmlFor={undefined}
      />
      <div
        className="c6"
      >
        <input
          onBlur={[Function]}
          onChange={[Function]}
          onKeyDown={[Function]}
          value=""
        />
        <i
          className="fa fa-times-circle c7"
        />
      </div>
      <span
        className="c8"
      />
    </div>
  </div>
  <div
    className="c9 c10"
    title="Filters"
  >
    <div
      className="dropdown-toggle c11"
      onClick={[Function]}
    >
      <div
        className="c12 c13"
      >
        Filters
      </div>
      <div>
        <span
          className="fa fa-caret-down c14"
        />
      </div>
    </div>
  </div>
</div>
`;
